### PR TITLE
Further fixes

### PR DESCRIPTION
**Pull request by mkrecek234**
- renderView protected
- and many other refactorings

### CODE REVIEW

**Comment by mkrecek234** [comment]
Solves issue #16

File: src/Field/File.php [hunk 1]
```diff
@@ -4,14 +4,14 @@
 
 namespace atk4\filestore\Field;
 
-class File extends \atk4\data\Field_SQL
+class File extends \atk4\data\FieldSQL
```
Comment (mvorisek) [inline src/Field/File.php:7]: `FieldSql` (notice the case)

File: src/Field/File.php [hunk 2]
```diff
@@ -4,14 +4,14 @@
...
 {
     use \atk4\core\InitializerTrait {
         init as _init;
     }
 
 
-    public $ui = ['form'=>'\atk4\filestore\FormField\Upload'];
+    public $ui = [['form'=>'\atk4\filestore\Form\Control\Upload']];
```
Comment (mvorisek) [inline src/Field/File.php:14]: use ::class (but same in reality...) and I think it should be more `['form' => [cl]]` (from seed definition, I do not know this plugin...)

File: src/Form/Control/Upload.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        $this->setFileId($f->get('token'));
+ 
+    }
+
+    public  function deleted($token)
+    {
+        $f = $this->field->model;
+        $f->tryLoadBy('token', $token);
+
+        $js =  new \atk4\ui\jsNotify(['content' => $f->get('meta_filename').' has been removed!', 'color' => 'green']);
```
Comment (mvorisek) [inline src/Form/Control/Upload.php:69]: `JsNotify`

File: src/Form/Control/Upload.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
+<?php
+namespace atk4\filestore\Form\Control;
+
+class Upload extends \atk4\ui\Form\Control\Upload
```
Comment (mvorisek) [inline src/Form/Control/Upload.php:4]: new file, isn't solved somewhere else?

File: src/Form/Control/Upload.php [hunk 5]
```diff
@@ -0,0 +1,77 @@
+<?php
+namespace atk4\filestore\Form\Control;
+
+class Upload extends \atk4\ui\Form\Control\Upload 
+{
+
+    public $model = null; // File model
```
Comment (mvorisek) [inline src/Form/Control/Upload.php:7]: annotate with phpdoc

File: src/Form/Control/Upload.php [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        }
+        parent::renderView();
+    }
+
+    public function uploaded($file)
+    {
+       
+        // provision a new file for specified flysystem
+        $f = $this->field->model;
+        $f->newFile($this->field->flysystem);
```
Comment (mvorisek) [inline src/Form/Control/Upload.php:34]: I belive I can fly... :)

File: src/Field/File.php (hunk 2 above)
Comment (DarkSide666) [reply src/Field/File.php:14]: Should be `['form' => [\atk4\filestore\Form\Control\Upload::class]]`

File: src/Form/Control/Upload.php (hunk 4 above)
Comment (mkrecek234) [reply src/Form/Control/Upload.php:4]: Sure, just moved the old Upload.php into the new folder structure

File: src/Form/Control/Upload.php [hunk 7]
```diff
@@ -10,7 +10,7 @@ function init(): void {
         parent::init();
 
         $this->onUpload(function ($files) {
-            $this->uploaded($files); 
+            $this->uploaded($file);
```
Comment (mvorisek) [inline src/Form/Control/Upload.php]: must be typo

File: src/Form/Control/Upload.php (hunk 4 above)
Comment (mvorisek) [reply src/Form/Control/Upload.php:4]: moved, really? I se a new file and no deletion...

**Comment by mkrecek234** [comment]
Whoever merges, it would be advisable to put in a tag, so that older versions are not overwritten. For the additional repos like filestore, login or the likes I see an issue that they all go into master

File: src/Form/Control/Upload.php (hunk 7 above)
Comment (mkrecek234) [reply src/Form/Control/Upload.php]: Was fixed above already. we should go for $files as in the original Upload class. This should be capable of handling multiple file uploads.

File: src/Form/Control/Upload.php (hunk 4 above)
Comment (DarkSide666) [reply src/Form/Control/Upload.php:4]: it's moved

**Pull request merged**